buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -286,11 +286,11 @@ public ByteBuf ensureWritable(int minWritableBytes) {
     final void ensureWritable0(int minWritableBytes) {
         final int writerIndex = writerIndex();
         final int targetCapacity = writerIndex + minWritableBytes;
-        if (targetCapacity <= capacity()) {
+        if (targetCapacity >= 0 & targetCapacity <= capacity()) {
```

Comment:
Can we please add a comment that `&` is really what we want here and not `&&` ? This will make it easier in the future to not get confused by this.

Reply:
Why use `&` instead of `&&` ?

Reply:
added a comment with rationale